include/curl/curl.h

```diff
@@ -2375,6 +2375,9 @@ CURL_EXTERN char *curl_easy_escape(CURL *handle,
 CURL_EXTERN char *curl_escape(const char *string,
                               int length);
 
+CURL_EXTERN char *curl_easy_escape_form(CURL *handle,
+                                        const char *string,
+                                        int length);
```

Comment:
I'm not convinced we want a new public API for this, and if we do, we need docs and more tests. But yes, I realize that if it isn't a new public API we need to figure how to get this done in another way.